src/ipc/ipc.c

```diff
@@ -309,10 +311,20 @@ int ipc_buffer_free(struct ipc *ipc, uint32_t buffer_id)
 	if (!cpu_is_me(ibd->core))
 		return ipc_process_on_core(ibd->core);
 
-	/* check state */
-	if (ibd->cb->sink->state != COMP_STATE_READY ||
-	    ibd->cb->source->state != COMP_STATE_READY)
-		return -EINVAL;
+	/* for each component */
```

Comment:
what are we doing here for each componet ? can you say in the comment.

Reply:
comment updated